great_expectations/core/batch_definition.py

```diff
@@ -83,9 +85,24 @@ def get_batch(self, batch_parameters: Optional[BatchParameters] = None) -> Batch
         return batch_list[-1]
 
     def is_added(self) -> BatchDefinitionAddedDiagnostics:
-        return BatchDefinitionAddedDiagnostics(
+        diagnostics = BatchDefinitionAddedDiagnostics(
             errors=[] if self.id else [BatchDefinitionNotAddedError(name=self.name)]
         )
+        if not diagnostics.is_added:
+            return diagnostics
+        return self._is_fresh()
```

Comment:
I think we should reverse `_is_fresh` and `is_added`. That is, the fresh check should verify that it's added and that it is fresh.